aggregator/options.go

```diff
@@ -0,0 +1,515 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package aggregator
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/m3db/m3aggregator/aggregation/quantile/cm"
+	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3metrics/protocol/msgpack"
+	"github.com/m3db/m3x/clock"
+	"github.com/m3db/m3x/instrument"
+)
+
+var (
+	defaultMetricPrefix           = []byte("stats.")
+	defaultCounterPrefix          = []byte("counts.")
+	defaultTimerPrefix            = []byte("timers.")
+	defaultTimerSumSuffix         = []byte(".sum")
+	defaultTimerSumSqSuffix       = []byte(".sum_sq")
+	defaultTimerMeanSuffix        = []byte(".mean")
+	defaultTimerLowerSuffix       = []byte(".lower")
+	defaultTimerUpperSuffix       = []byte(".upper")
+	defaultTimerCountSuffix       = []byte(".count")
+	defaultTimerStdevSuffix       = []byte(".stdev")
+	defaultTimerMedianSuffix      = []byte(".median")
+	defaultGaugePrefix            = []byte("gauges.")
+	defaultMinFlushInterval       = 5 * time.Second
```

Comment:
This should probably always be the `policy resolution / 2` right?  Perhaps each list should have a different one that can default to this but should be at the very least `policy resolution / 2`?

Otherwise how does the 1s lists get flushed fast enough.